### PR TITLE
#10567: Remove std::function for bw ops

### LINKED ISSUE #10567: Remove std::function for next set of ops
Tracking Issue : #10564
List of ops to cover: #10698

- [x]  logaddexp2_bw
- [x] squared_difference_bw
- [x] max_bw
- [x] min_bw

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #10567
https://github.com/tenstorrent/tt-metal/actions/runs/10082687977 - passed

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.hpp [hunk 1]
```diff
@@ -121,6 +125,34 @@ struct OpHandler_binary_bw<BinaryBackwardOpType::LOGADDEXP_BW> {
     }
 };
 
+template <>
+struct OpHandler_binary_bw<BinaryBackwardOpType::LOGADDEXP2_BW> {
+    static std::vector<Tensor> handle( const Tensor& grad, const Tensor& input, const Tensor& other, const std::optional<MemoryConfig>& output_mem_config ) {
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.hpp:130]: Please fix this.

**Review by eyonland — CHANGES_REQUESTED** [review]
https://github.com/tenstorrent/tt-metal/pull/10710

**Review by TT-BrianLiu — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.hpp (hunk 1 above)
Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.hpp:128]: Issue fixed in PR #10725

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 2]
```diff
@@ -248,21 +236,22 @@ other: grad / (1 + pow(2, self - other))
 */
 
 std::vector<ttnn::Tensor> _logaddexp2_bw(
-    const Tensor& grad, const Tensor& input_a, const Tensor& other, const MemoryConfig& output_mem_config) {
+    const Tensor& grad, const Tensor& input_a, const Tensor& other, const std::optional<MemoryConfig>& output_mem_config) {
     std::vector<Tensor> grad_tensor;
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:240]: At some point we need to change output_mem_config to just memory_config

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary/complex_unary.hpp [hunk 3]
```diff
@@ -20,17 +20,15 @@ struct ExecuteComplexUnaryType1 {
...
-        auto op_type = get_function_complex_unary<complex_unary_op_type>();
-        return op_type(input_tensor_arg, memory_config);
+        return OpHandler<complex_unary_op_type>::handle(input_tensor_arg, memory_config);
     }
 };
 
 //OpHandler_complex_type2 = get_function_complex_unary_type2 --> ComplexTensor return type
 template <ComplexUnaryOpType complex_unary_op_type>
 struct ExecuteComplexUnaryType2 {
     static ComplexTensor operator()(const ComplexTensor &input_tensor_arg, const MemoryConfig &memory_config) {
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary/complex_unary.hpp:30]: Please remove Type2

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp [hunk 4]
```diff
@@ -1603,18 +1608,6 @@ std::vector<Tensor> _prod_bw(
 
 std::function<std::vector<ttnn::Tensor>(const Tensor&, const Tensor&, const MemoryConfig&)> UnaryBackwardFunction::get_function_type1(UnaryBackwardOpType OpType){
     switch (OpType) {
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp:1610]: Please remove get_function_type1

**Review by eyonland — APPROVED** [review]
Please remove get_function_type1 either on this PR or another.

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10114444391 - passed

**Pull request merged**